core/controllers/profile_test.py

```diff
@@ -858,6 +859,9 @@ def test_export_account_handler(self):
         time_swap = self.swap(
             user_services, 'record_user_logged_in', lambda *args: None)
 
+        os.environ['TZ'] = 'UTC'
```

Comment:
No, don't do this -- it's messy and too ad hoc. It also absolutely could have an effect on other tests, so I can't say for certain that the new backend test failure is not due to your changes.

Could you please try and explain, in detail, why and in what contexts the existing backend test is failing? What is the flow of logic and the series of actions that results in the error? I need to understand that in more detail before I can suggest the right way to fix it.

Reply:
Here is the issue:

The timezone used by the CI is UTC; however, the timezones of contributor computers could vary. The python utility function (that is based on the datetime library) will use the timezone local to the computer for generating the epoch timestamp, rather than using a UTC epoch. As a result, when contributors run the tests locally, the tests will fail because the timestamps generated by the export_data function will not match the timestamps in the zip file (which are UTC timestamps for the CI).

Thus, the rationale behind this change is that in testing, it "normalizes" the timezone to be UTC.

Reply:
Thanks -- great explanation. In that case, though, this should be part of the generic setUp() and tearDown() in AppEngineTestBase in test_utils.py.

In other words: store the current env var, set it to the new thing in setUp(), and set it back to the old thing in tearDown(). Add an explanation there, too (similar to the one you gave above). That is a tidier way to do it, and it will apply neatly to all the tests on all envs.

Reply:
Got it. Done.